python/lsst/pipe/tasks/visualizeVisit.py

```diff
@@ -0,0 +1,250 @@
+    """
+    binning = pexConfig.Field(
+        dtype=int,
+        default=8,
+        doc="Binning factor to apply.",
+    )
+
+
+class VisMosaicTask(pipeBase.PipelineTask,
+                    pipeBase.CmdLineTask):
+    """Task to mosaic binned products.
+    """
+    ConfigClass = VisMosaicConfig
+    _DefaultName = 'VisMosaic'
+
+    def makeCameraImage(self, inputExps, camera, binning):
+        """Make an image of an entire focal plane.
+
+        Parameters
+        ----------
+        exposures: `dict` [`int`, `lsst.afw.image.Exposure`]
+            CCD exposures, binned by `binning`.  The keys are the
+            detectorIDs, with the values the binned image exposure.
+
+        Returns
+        -------
+        image : `lsst.afw.image.Image`
+            Image mosaicked from the individual binned images for each
+            detector.
+        """
+        class ImageSource:
+            """Source of images for makeImageFromCamera"""
+            def __init__(self, exposures):
+                self.isTrimmed = True
+                self.exposures = exposures
+                self.background = np.nan
+
+            def getCcdImage(self, detector, imageFactory, binSize):
+                """Provide image of CCD to makeImageFromCamera
+
+                Parameters
+                ----------
+                detector : `int`
+                    Detector ID to get image data for.
+                imageFactory : `lsst.afw.image.Image`
+                    Type of image to construct.
+                binSize : `int`
+                    Binsize to use to recompute dimensions.
+
+                Returns
+                -------
+                image : `lsst.afw.image.Image`
+                    Appropriately rotated, binned, and transformed
+                    image to be mosaicked.
+                detector : `lsst.afw.cameraGeom.Detector`
+                    Camera detector that the returned image data
+                    belongs to.
+                """
+                detId = detector.getId()
+                if detId not in self.exposures:
+                    dims = detector.getBBox().getDimensions()/binSize
+                    image = imageFactory(*[int(xx) for xx in dims])
+                    image.set(self.background)
+                else:
+                    image = self.exposures[detector.getId()]
+                if hasattr(image, "getMaskedImage"):
+                    image = image.getMaskedImage()
+                if hasattr(image, "getMask"):
+                    mask = image.getMask()
+                    isBad = mask.getArray() & mask.getPlaneBitMask("NO_DATA") > 0
+                    image = image.clone()
+                    image.getImage().getArray()[isBad] = self.background
+                if hasattr(image, "getImage"):
+                    image = image.getImage()
+
+                image = afwMath.rotateImageBy90(image, detector.getOrientation().getNQuarter())
```

Comment:
You should check if they actually need rotated or put a comment saying that rotateImageBy90 does the check, It would be a shame to do lots of linear algebra with identity matrices